integrationtest/showcase/library/src/main/java/org/hibernate/search/integrationtest/showcase/library/model/Document.java

```diff
@@ -64,6 +64,29 @@
 	@IndexedEmbedded(includePaths = {"medium", "library.location", "library.services"}, storage = ObjectFieldStorage.NESTED)
 	private List<C> copies = new ArrayList<>();
 
+	public Document() {
+	}
+
+	public Document(int id, String title, String author, String summary, String tags) {
+		this.id = id;
+		this.title = title;
+		this.author = author;
+		this.summary = summary;
+		this.tags = tags;
+	}
+
+	public C getCopy(Library library, int copyIndex) {
```

Comment:
In general, I prefer to push the logic inside the entity in order to avoid an [Anemic Domain Model](https://www.martinfowler.com/bliki/AnemicDomainModel.html). Furthermore, some people say that with Hibernate it is impossible to build a model avoiding an Anemic Domain and this would be a good counterexample example of that. Anyway let's do as you wish ;)